jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/sender/JMSClientConnector.java

```diff
@@ -122,12 +124,14 @@ public synchronized boolean send(CarbonMessage carbonMessage, CarbonCallback car
         } catch (JMSException e) {
             throw new ClientConnectorException("Exception occurred while sending the message", e);
         } finally {
-            try {
-                jmsConnectionFactory.closeMessageProducer(messageProducer);
-                jmsConnectionFactory.closeSession(session);
-                jmsConnectionFactory.closeConnection(connection);
-            } catch (JMSConnectorException e) {
-                throw new ClientConnectorException(e.getMessage(), e);
+            if (jmsConnectionFactory != null) {
+                try {
+                    jmsConnectionFactory.closeMessageProducer(messageProducer);
+                    jmsConnectionFactory.closeSession(session);
+                    jmsConnectionFactory.closeConnection(connection);
+                } catch (JMSConnectorException e) {
+                    logger.error("Exception occured when closing connection " + e.getMessage());
```

Comment:
"e" is lost here. You have to include the "e" to get the stack trace.

Reply:
This is fixed.